orttraining/orttraining/training_api/onnxruntime_training_c_api.cc

```diff
@@ -491,6 +533,7 @@ static constexpr OrtTrainingApi ort_training_api = {
     &OrtTrainingApis::LoadCheckpoint,
     &OrtTrainingApis::SaveCheckpoint,
     &OrtTrainingApis::CreateTrainingSession,
+    &OrtTrainingApis::CreateTrainingSessionFromArray,
```

Comment:
May we add new functions to the end of the struct?

Reply:
Wanted to keep similar apis together. Right now since training is very new, I thought it was OK to do this. Later once we have adoption this wont be an option...

Reply:
Resolving per offline conversation. We decided it is OK to make such a change given on-device training is very new and we dont have any production usage yet.